apps/linkedin-to-notion/src/contents/linkedin-profile-scraper.ts

```diff
@@ -0,0 +1,126 @@
+import type { PlasmoCSConfig } from 'plasmo';
+
+import { cleanJobTitle, splitName } from './linkedin-data-cleaners';
+
+export const config: PlasmoCSConfig = {
+  matches: ['https://www.linkedin.com/in/*'],
+};
+
+type LinkedInProfileInformation = {
+  name: {
+    firstName: string;
+    lastName: string;
+  };
+  jobTitle: string;
+  currentCompany: string;
+  location: string;
+  linkedInURL: string;
+};
+
+const getName = (): {
+  firstName?: string;
+  lastName?: string;
+  fullName: string;
+} => {
+  const detailsContainer = document.querySelector('.pv-text-details__left-panel');
+  if (!detailsContainer) {
+    return {
+      fullName: '',
+    };
+  }
+
+  const name = detailsContainer.querySelector('h1.text-heading-xlarge');
+  if (!name) {
+    return {
+      fullName: '',
+    };
+  }
+
+  const { firstName, lastName } = splitName(name.textContent);
+
+  return {
+    fullName: name.textContent,
+    firstName,
+    lastName,
+  };
+};
+
+const getJobTitle = (): string => {
+  const experienceAnchor = document.getElementById('experience');
+  if (!experienceAnchor) {
+    return '';
+  }
+
+  const experienceList = experienceAnchor.parentElement;
+  if (!experienceList) {
+    return '';
+  }
+
+  const lastJob = experienceList.querySelector('li');
+  if (!lastJob) {
+    return '';
+  }
+
+  const jobTitle = lastJob.querySelector('div.display-flex.align-items-center.mr1.t-bold > span');
+  if (!jobTitle) {
+    return '';
+  }
+
+  return cleanJobTitle(jobTitle.textContent);
+};
```

Comment:
I reckon you only want to get the job title from the experience's list. I agree that it's the most reliable but you might face an issue: the elements **won't be loaded in the page** when landing on it, unless you scroll farther down.

That's why I'd add the following last resort:
```
document.querySelector("div.text-body-medium.break-words").innerText.trim();
```

Reply:
Indeed! I added a fallback but I still think that this can be largely optimised. I don't think it's the intent of the project but we'll have to think about reliability of this scraper and how to test it.